aten/src/ATen/native/EmbeddingBag.cpp

```diff
@@ -59,6 +64,59 @@ static void index_select_add(const Tensor &select_indices,
   }
 }
 
+template<>
+void index_select_add<float>(const Tensor &select_indices,
+                             const Tensor &add_indices,
+                             const Tensor &src,
+                             Tensor &output,
+                             const Tensor& offsets) {
+  AT_ASSERT(select_indices.numel() == add_indices.numel());
+  auto add_indices_data = add_indices.data<int64_t>();
+  auto select_indices_data = select_indices.data<int64_t>();
+  auto src_data = src.data<float>();
+  auto output_data = output.data<float>();
+  auto numel = add_indices.numel();
+  int64_t ddim = src.size(1);
+  auto src_stride0 = src.stride(0);
+  auto src_stride1 = src.stride(1);
+  auto output_stride0 = output.stride(0);
+  auto output_stride1 = output.stride(1);
+
+  // We specialize the case where both the src and output are contiguous in the
+  // last dimension to give the compiler a better chance to vectorize.
+  if (src_stride1 == 1 && output_stride1 == 1) {
+    auto accessor = offsets.accessor<int64_t, 1>();
+    std::vector<int> lengths;
+
+    int64_t lower = accessor[0];
```

Comment:
it assumes the offsets is non-empty. Probably usually the case, but better to work on empty batch too (seems like we'd need to fix functional.py too)